### PR TITLE
Add basic CQL + SQL language

### PR DESCRIPTION
**Pull request by lytefast**

### CODE REVIEW

**Comment by lytefast** [comment]
@jkcailteux This is an example of how you could add a language

**Comment by jkcailteux** [comment]
> @jkcailteux This is an example of how you could add a language

Could you provide a few more examples?

**Comment by lytefast** [comment]
https://github.com/discord/SimpleAST/commit/6690ce24e1ffbe1c0340d805da3eb5e85078e306 is the addition for XML; which has some custom Regex for tags + comments.

but if you look at https://github.com/discord/SimpleAST/blob/f3c51420d98a54ba6ed3d9a0ec928eca1af3fb5c/simpleast-core/src/main/java/com/discord/simpleast/code/CodeRules.kt#L106-L142 for more examples.

I'm not sure what you are specifically looking for. but if you really want custom stuff you'll have to make regex parsers.

File: simpleast-core/src/main/java/com/discord/simpleast/code/CodeRules.kt [hunk 1]
```diff
@@ -172,6 +172,23 @@ object CodeRules {
         PATTERN_TEXT.toMatchGroupRule(),
     )
 
+    val queryLangauge = listOf<Rule<R, Node<R>, S>>(
+        createSingleLineCommentPattern("#")
+            .toMatchGroupRule(stylesProvider = codeStyleProviders.commentStyleProvider),
+        Pattern.compile("""^"[\s\S]*?(?<!\\)"(?=\W|\s|$)""")
+            .toMatchGroupRule(stylesProvider = codeStyleProviders.literalStyleProvider),
+        createWordPattern("true|false|null").pattern().toPattern(Pattern.CASE_INSENSITIVE)
```
Comment (lytefast) [inline simpleast-core/src/main/java/com/discord/simpleast/code/CodeRules.kt:180]: this is a small hack so I can reuse the `createWordPattern` regex but add an option.
it only happens on creation so should be low cost overall.

**Review by AndyG — APPROVED** [review]

**Pull request merged**